### PR TITLE
For main functions returning a Future, translate into JS Promises when creating an npm package

### PR DESCRIPTION
**Pull request by BobobUnicorn**
This addresses #50 and enables proper error reporting in rules_sass in worker mode (bazelbuild/rules_sass#96).

I am not entirely sure if this warrants a particular unit test, given that this is basically a private API.

### CODE REVIEW

File: lib/src/npm.dart [hunk 1]
```diff
@@ -296,6 +296,29 @@ String get _wrapperLibrary {
...
+@JS()
+class Promise<T> {
+  external Promise(void executor(void resolve(T result), Function reject));
+}
+
+Promise<T> _futureToPromise<T>(Future<T> future) {
+  return new Promise<T>(allowInterop((resolve, reject) {
+    future.then(resolve, onError: reject);
+  }));
+}
```
Comment (nex3) [inline lib/src/npm.dart:311]: Rather than writing these ourselves, let's use them from [`node_interop`](https://pub.dev/packages/node_interop).

**Review by nex3 — CHANGES_REQUESTED** [review]

File: lib/src/npm.dart (hunk 1 above)
Comment (BobobUnicorn) [reply lib/src/npm.dart]: sg; done

File: lib/src/npm.dart [hunk 2]
```diff
@@ -296,6 +297,24 @@ String get _wrapperLibrary {
...
   }
 
+  // Define a JS-interop Future to Promise translator so that we can export
+  // a Promise-based API
+  wrapper.writeln("""
+Promise _futureToPromise(Future future) {
+  return new Promise(allowInterop((resolve, reject) {
+    future.then(resolve, onError: reject);
+  }));
+}
```
Comment (nex3) [inline lib/src/npm.dart:307]: Why not use [`futureToPromise()`](https://pub.dev/documentation/node_interop/latest/node_interop.util/futureToPromise.html) from `node_interop`?

Comment (BobobUnicorn) [reply lib/src/npm.dart]: Somehow, I missed finding that. (done)

**Comment by nex3** [comment]
You'll need to merge with master to fix the build error, and you'll need to bump the pubspec to 1.1.0 and add a changelog entry. (I'd do this myself but it looks like you've disabled allowing me to push to your PR branch.)

**Comment by BobobUnicorn** [comment]
Sorry, I've been busy for a bit - I am not sure if this is considered a breaking change (and so would be v2.0.0)

**Review by nex3 — APPROVED** [review]

**Pull request merged**